### PR TITLE
Core 106 evolutionary algorithm for strategies

### PR DESCRIPTION
**Pull request by birchmd**

### CODE REVIEW

**Comment by ghost** [comment]
Please add a couple paragraphs of description and any places where you'd like the reviewer to concentrate their review efforts.

**Comment by birchmd** [comment]
-Core of the evolutionary algorithm is in Generation.nextGeneration method
-Core of the network model is in UniformRandomDelay
-Current strategy for validation is ThresholdSpender
-packages protocol and block capture what I wrote on the wiki page
-everything else is helper functions and capturing data about the simulations for later analysis

File: simulation/src/main/scala/coop/rchain/caspersimulation/strategy/evolution/Generation.scala [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    reporter.record(network) //record final state
+
+    val fit = validators.map(v => v -> fitness(v)).toMap
+    val mostFitValidator = fit.maxBy(_._2)._1
+    val aveFitness = fit.valuesIterator.sum / populationSize
+    val numChildren = fit.mapValues(f => round(f / aveFitness).toInt)
+    val newPopSize = numChildren.valuesIterator.sum
+    val newValidators = numChildren.flatMap{
+      case (v, nc) =>
+        val childGenerator = if (v == mostFitValidator) { //give most fit validator extra kids to prevent population decrease
```
Comment (ghost) [inline simulation/src/main/scala/coop/rchain/caspersimulation/strategy/evolution/Generation.scala]: Move the comment down a line so the line isn't so wide

**Review by ghost — COMMENTED** [review]
Looks great.  Fix the one nit and I'll approve.

**Comment by birchmd** [comment]
Thanks Mike :)

**Review by ghost — APPROVED** [review]

**Pull request merged**